api/server/prom_zip_collector.go

```diff
@@ -0,0 +1,80 @@
+package server
+
+import (
+	"github.com/fnproject/fn/api/agent"
+	"github.com/openzipkin/zipkin-go-opentracing"
+	"github.com/openzipkin/zipkin-go-opentracing/thrift/gen-go/zipkincore"
+	"github.com/prometheus/client_golang/prometheus"
+	"strings"
+)
+
+// Each span name is published as a separate Histogram metric
+// Using metric names of the form fn_span_<span-name>_duration_seconds
+var histogramVecMap = make(map[string]*prometheus.HistogramVec)
+
+// Return the HistogramVec corresponding to the specified spanName.
+// If a HistogramVec does not already exist for specified spanName then one is created and configured with the specified labels
+// otherwise the labels parameter is ignored.
+func getHistogramVecForSpanName(spanName string, labels []string) *prometheus.HistogramVec {
+	thisHistogramVec, found := histogramVecMap[spanName]
+	if !found {
+		thisHistogramVec = prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Name: "fn_span_" + spanName + "_duration_seconds",
+				Help: "Span " + spanName + " duration, by span name",
+			},
+			labels,
+		)
+		histogramVecMap[spanName] = thisHistogramVec
+		prometheus.MustRegister(thisHistogramVec)
+	}
+	return thisHistogramVec
+}
+
+// PrometheusCollector is a custom Collector
+// which sends ZipKin traces to Prometheus
+type PrometheusCollector struct {
+	a agent.Agent
```

Comment:
this field seems unused?

Reply:
yeah, it does seem agent is not used anywhere.

Reply:
Good spot. Removed.